server/asset/eth/coiner.go

```diff
@@ -197,78 +186,11 @@ func (be *AssetBackend) baseCoin(coinID []byte, contractData []byte) (*baseCoin,
 // and the same account and nonce, effectively voiding the transaction we
 // expected to be mined.
 func (c *swapCoin) Confirmations(ctx context.Context) (int64, error) {
-	swap, err := c.backend.node.swap(ctx, c.backend.assetID, c.secretHash)
-	if err != nil {
-		return -1, err
-	}
-
-	// Uninitiated state is zero confs. It could still be in mempool.
```

Comment:
I've moved these functions away from contract inspection and towards more transaction inspection. I think it simplifies things. Any reason I can't do that?

Reply:
I wanted to ensure that this info comes from the contract and not our assumptions about the tx. Especially with the recent redemptions pr we may be looking at a bogus transaction. (I don't think this is used for redemptions, but maybe it will be some day.)

If you are sure we are not then fine, but I'm not sure we couldn't have a tx that passed our verification but does not do what we expect. **Our txdata parsing is basically a hack**, we had to copy paste from geth internals. I do not think it is intended to be done this way.

At least there needs to be another check besides confirmations that checks status is ok, as in it did what we expected onchain. But, I hope we can not trust the transaction except for the basic sanity check we were doing and start looking at the contract here.

Reply:
It's certainly not the eth way.  Things are meant to replaceable, and the focus is supposed to be on state change, and contract interactions.

I'd feel better about doing less tx inspection with the hacked call data parsing if possible.

I worry we're already dug in on something we shouldn't be doing as well.